src/answer/answer.service.ts

```diff
@@ -135,30 +175,31 @@ export class AnswerService {
 
   async isFavorite(
     answerId: number,
-    userId: number | undefined,
+    createdById: number | undefined,
   ): Promise<boolean> {
-    if (userId == undefined) return false;
+    if (createdById == undefined) return false;
     const answer = await this.answerRepository.findOne({
       where: { id: answerId },
       relations: ['favoritedBy'],
     });
     if (!answer) {
       throw new AnswerNotFoundError(answerId);
     }
-    return answer.favoritedBy.some((user) => user.id === userId);
+    return answer.favoritedBy.some((user) => user.id === createdById);
   }
 
   async getAnswerDto(
+    questionId: number,
```

Comment:
Well, I think this can be checked in the controller layer by calling a method `isAnswerMatchQuestion` exposed in the service

Reply:
And every endpoint like `questions/{id}/answers/{id}` need to check it